extensions/data-workspace/src/services/workspaceService.ts

```diff
@@ -13,11 +14,30 @@ import * as constants from '../common/constants';
 
 const WorkspaceConfigurationName = 'dataworkspace';
 const ProjectsConfigurationName = 'projects';
+const TempProject = 'tempProject';
 
 export class WorkspaceService implements IWorkspaceService {
 	private _onDidWorkspaceProjectsChange: vscode.EventEmitter<void> = new vscode.EventEmitter<void>();
 	readonly onDidWorkspaceProjectsChange: vscode.Event<void> = this._onDidWorkspaceProjectsChange?.event;
 
+	constructor(private _context: vscode.ExtensionContext) {
+		this.loadProject();
```

Comment:
this should be the logic in the activate method of the extension, right? workspace service should provide the methods to read and write the temp project, but extension activation should be the one to invoke it.

Reply:
makes sense. I moved it to get called in the extension activate()